File: software/examples/php/ExampleSimple.php
<?php

require_once('Tinkerforge/IPConnection.php');
require_once('Tinkerforge/BrickletSolidStateRelay.php');

use Tinkerforge\IPConnection;
use Tinkerforge\BrickletSolidStateRelay;

const HOST = 'localhost';
const PORT = 4223;
const UID = 'XYZ'; // Change XYZ to the UID of your Solid State Relay Bricklet

$ipcon = new IPConnection(); // Create IP connection
$ssr = new BrickletSolidStateRelay(UID, $ipcon); // Create device object

$ipcon->connect(HOST, PORT); // Connect to brickd
// Don't use device before ipcon is connected

// Turn relay on/off 10 times with 1 second delay
for($i = 0; $i < 5; $i++) {
    sleep(1);
    $ssr->setState(TRUE);
    sleep(1);
    $ssr->setState(FALSE);
}

echo "Press key to exit\n";
fgetc(fopen('php://stdin', 'r'));
$ipcon->disconnect();

?>

File: software/examples/ruby/example_simple.rb
#!/usr/bin/env ruby
# -*- ruby encoding: utf-8 -*-

require 'tinkerforge/ip_connection'
require 'tinkerforge/bricklet_solid_state_relay'

include Tinkerforge

HOST = 'localhost'
PORT = 4223
UID = 'XYZ' # Change to your UID

ipcon = IPConnection.new # Create IP connection
ssr = BrickletSolidStateRelay.new UID, ipcon # Create device object

ipcon.connect HOST, PORT # Connect to brickd
# Don't use device before ipcon is connected

# Turn relay on/off 10 times with 1 second delay
for _ in 0..4
  sleep 1
  ssr.set_state true
  sleep 1
  ssr.set_state false
end

puts 'Press key to exit'
$stdin.gets
ipcon.disconnect
